### PR TITLE
DSP-3751: tsv and parquet logs must be created together

### PR DESCRIPTION
**Pull request by nofateg**

### CODE REVIEW

File: oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java [hunk 1]
```diff
@@ -132,19 +153,26 @@ public void log( ProtocolVersion protocolVersion, String hostName, String filePr
...
+                }
 
-            log.error( "hostName {} filePrefix {} logType {} properties {} shard {} headers {} path {}",
-                hostName, filePreffix, logType, properties, shard, headersWithTypes, writer.currentPattern() );
-            throw e;
+                log.error( "hostName {} filePrefix {} logType {} properties {} shard {} headers {} path {}",
+                    hostName, filePreffix, logType, properties, shard, headersWithTypes, writer.currentPattern() );
+                boolean isThereAlreadyRecordedLog = iw > 0;
+                if( !isThereAlreadyRecordedLog ) {
+                    throw e;
```
Comment (vdiravka) [inline oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java:173]: For one log file exception won’t be thrown, right?

File: oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java [hunk 2]
```diff
@@ -132,19 +153,26 @@ public void log( ProtocolVersion protocolVersion, String hostName, String filePr
...
-        var writer = writers.get( new LogId( filePreffix, logType, hostName, shard, properties, headers, types ) );
-        log.trace( "logging {} bytes to {}", length, writer );
-        try {
-            writer.write( protocolVersion, buffer, offset, length, this.listeners::fireError );
-        } catch( Exception e ) {
-            var headersWithTypes = new ArrayList<String>();
-            for( int i = 0; i < headers.length; i++ ) {
-                headersWithTypes.add( headers[i] + " [" + Lists.map( List.of( ArrayUtils.toObject( types[i] ) ), oap.template.Types::valueOf ) + "]" );
-            }
+        List<AbstractWriter<? extends Closeable>> writerList = writers.get( new LogId( filePreffix, logType, hostName, shard, properties, headers, types ) );
```
Comment (vdiravka) [inline oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java:156]: `writerList` for 2 formats, but `LogId` identifies only one `filePreffix` and `logType`.
Should it be changed? Possibly 2 `LogId` should be transferred

File: oap-logstream/src/main/java/oap/logstream/disk/LogFormat.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package oap.logstream.disk;
+
+public enum LogFormat {
+    TSV_GZ( "tsv.gz" ),
```
Comment (vdiravka) [inline oap-logstream/src/main/java/oap/logstream/disk/LogFormat.java:28]: `tsv` with compression by default. What about parquet?

File: oap-logstream/src/main/java/oap/logstream/disk/ParquetWriter.java [hunk 4]
```diff
@@ -176,19 +187,24 @@ private void convertToParquet( byte[] buffer, int offset, int length, byte[][] t
...
+                if( !excludeFields.contains( header ) ) {
+                    try {
+                        addValue( parquetCol, obj, colType, 0, group );
+                    } catch( Exception e ) {
+                        log.error( "header {} class {} type {} col {}", header, obj.getClass().getName(),
+                            Lists.map( List.of( ArrayUtils.toObject( types[col] ) ), oap.template.Types::valueOf ),
+                            parquetCol );
+
+                        var data = BinaryUtils.read( buffer, offset, length );
+                        log.error( "object data {}", data );
```
Comment (vdiravka) [inline oap-logstream/src/main/java/oap/logstream/disk/ParquetWriter.java:203]: why not combined with an above log.error?

**Review by vdiravka — COMMENTED** [review]
Is there test with logging 2 files simultaneously?

File: oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java (hunk 1 above)
Comment (vdiravka) [reply oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java:173]: Got `YES` answer offline

File: oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java (hunk 2 above)
Comment (nofateg) [reply oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java:156]: LogId#logType is not a log file format, it is a data format (bids, impressions)

File: oap-logstream/src/main/java/oap/logstream/disk/LogFormat.java (hunk 3 above)
Comment (nofateg) [reply oap-logstream/src/main/java/oap/logstream/disk/LogFormat.java:28]: Parquet is a binary format and compression is an internal configuration.

File: oap-logstream/src/main/java/oap/logstream/disk/ParquetWriter.java (hunk 4 above)
Comment (nofateg) [reply oap-logstream/src/main/java/oap/logstream/disk/ParquetWriter.java:203]: This is an additional log to make troubleshooting easier.

**Pull request merged**